wcfsetup/install/files/lib/http/Helper.class.php

```diff
@@ -23,6 +26,30 @@ public static function isAjaxRequest(ServerRequestInterface $request): bool
         return $request->getHeaderLine('x-requested-with') === 'XMLHttpRequest';
     }
 
+    /**
+     * Validates source parameters against the provided schema. Unknown
+     * keys are skipped and values are gracefully converted into the
+     * requested types.
+     *
+     * The returned array will contain only the values specified in the
+     * schema. Missing parameters or values that cannot be casted to the
+     * requested type will yield a `MappingError`.
+     *
+     * @throws MappingError
+     */
+    public static function mapRequestParameters(array $sourceParameters, string $schema): mixed
```

Comment:
The implemented mapper is appropriate for query-string mapping and thus should be named as such. Within the request body we don't usually need to support superfluous keys and if JSON is used, flexible casting is not required either.